http-clients-api/src/main/java/com/palantir/remoting1/clients/ClientConfig.java

```diff
@@ -65,20 +60,12 @@ public Integer maxNumRetries() {
         return MAX_NUM_RETRIES;
     }
 
-    @Value.Check
-    public void check() {
-        Preconditions.checkState(sslSocketFactory().isPresent() == trustManager().isPresent(),
-                "Must set either both SslSocketFactory and TrustManager, or neither");
-    }
-
     public static ClientConfig fromServiceConfig(ServiceConfiguration serviceConfig) {
         ClientConfig.Builder clientConfig = builder();
 
         // ssl
         if (serviceConfig.security().isPresent()) {
-            clientConfig.sslSocketFactory(SslSocketFactories.createSslSocketFactory(serviceConfig.security().get()));
-            clientConfig.trustManager(
-                    (X509TrustManager) SslSocketFactories.createTrustManagers(serviceConfig.security().get())[0]);
+            clientConfig.trustContext(SslSocketFactories.createTrustContext(serviceConfig.security().get()));
```

Comment:
as a convenience. it's a thing that's not needed clients will likely always have an optional TrustContext depending on config.